### DISCUSSION TITLE
Open external links in the most recently used profile

### DISCUSSION THREAD

**Discussion by mcmerdith** [zen-browser/desktop#3023] [Ideas]
I used Chrome before switching to Zen, and when a link was opened by an external app Chrome would open it in the most recently used profile. Zen doesn't do this and always opens links in the default profile.

Usually this isn't a big issue but some applications do their sign in flows through "magic links" that open automatically in the default browser (for example Zoom). In the specific case of Zoom I sign in using an organization managed account that is only signed in on a non-default profile so to sign in I had to change the default profile to be the organization managed one and then change it back. Minor inconvenience, but it's definitely the "feature" I miss most from Chrome.

**Comment by aksdb** [comment]
I personally prefer being able to select the profile. So if it always uses the default profile (which I can change) it is "safer" to use than if it would use the one I had last active. Example: I have my work and private profile open at the same time. Depending on which window I had last active, Chrome will open external links in that session. So sometimes my work links open in my private session or vice versa. I dislike that. Microsoft added an option for that in Edge to specify in which profile to open external links in ... which happens to be more or less the current behavior of Zen, right?

What could be an improvement, if there is the option to _select_ the (currently running) profile to open external links in, whenever a link is opened. But even that should be just an option, IMO.

**Comment by mcmerdith** [comment]
I'd agree, an option would be a sensible middle-ground, because I definitely prefer the chrome way. I use my profiles roughly equally and having to change the default every time I want to open a link in a different profile feels a little clunky in my use case

**Comment by yyuyulm** [comment]
Having the ability to select profile in a right click panel would be great if possible. Or intercepting the link upon opening to allow user to select profile and workspace to open it in would be great too.

**Comment by arichyx** [comment]
It's an essential feature for business user. When I click an external link of my organization (zoom/slack/google doc), it always open in my personal profile instead of work profile, which is very annoying.

**Comment by rods012** [comment]
Also came looking to see if this has been suggested, I agree that if there was an option for people to choose if links open to a specific Profile or to the latest one it would be best.

I also wanted to link to a fork of the profile switcher for vanilla that has a toggle for this behavior (and that I have been using for the last half year with Firefox). Was very convenient!

https://github.com/Invertex/firefox-profile-switcher/releases/tag/v1.3.1-external-link-mod

**Comment by SheepDomination** [comment]
As mentioned in the original post, this is an issue; I want links to open in the most recently used profile. Zen just picks the default installed version of Zen.

**Comment by itsamenathan** [comment]
[Firefox 138.0](https://www.mozilla.org/en-US/firefox/138.0/releasenotes/) added the new profile management feature. It fixes this issue. The problem is that the profile management needs to be enabled via about:config by setting `browser.profiles.enabled` to true.

**Comment by phifogg** [comment]
Might be me or my setup... I am using Zen on apple with two profiles - work and private. Work is set on default for this machine, but still links open in the private profile. Can't really understand why. This is annoying for the reasons mentioned above.
